xbmc/addons/AddonManager.cpp

```diff
@@ -215,21 +218,38 @@ VECADDONS CAddonMgr::GetAvailableUpdates() const
   VECADDONS installed;
   VECADDONS all_addons;
   m_database.GetRepositoryContent(all_addons);
-  std::map<std::string, AddonPtr> last_versions;
+  std::map<std::string, AddonPtr> latest_private_versions;
+  std::map<std::string, AddonPtr> latest_trusted_versions;
   for (const auto& addon : all_addons)
   {
-    const auto& latest_known = last_versions.find(addon->ID());
-    if (latest_known == last_versions.end() || addon->Version() > latest_known->second->Version())
-      last_versions[addon->ID()] = std::move(addon);
+    if (IsFromTrustedRepo(addon))
+    {
+      AddAddonIfLatest(addon, latest_trusted_versions);
+    }
+    else
+    {
+      AddAddonIfLatest(addon, latest_private_versions);
+    }
   }
 
+  CLog::Log(LOGDEBUG, "ADDONS: trusted versions count : {}", latest_trusted_versions.size());
+  CLog::Log(LOGDEBUG, "ADDONS: private versions count : {}", latest_private_versions.size());
+
   GetAddonsForUpdate(installed);
+
   for (const auto& addon : installed)
   {
-    const auto& remote = last_versions.find(addon->ID());
-    if (remote != last_versions.end() && remote->second->Version() > addon->Version())
-      updates.emplace_back(std::move(remote->second));
+    CLog::Log(LOGDEBUG, "ADDONS: update check: addonID = {} / Origin = {}",
+        addon->ID(), addon->Origin());
+
+    // if the addon is not found in a trusted repo(s)...
+    if (!FindAddonAndCheckUpdate(addon, latest_trusted_versions, updates))
```

Comment:
think we have to handle `ORIGIN_SYSTEM` here too. otherwise system-addons won't get updated?